packages/mui-base/src/NumberInput/NumberInput.test.tsx

```diff
@@ -0,0 +1,469 @@
+      expect(handleValueChange.args[2][1]).to.equal(11);
+      expect(input.value).to.equal('11');
+    });
+
+    it('ArrowUp and ArrowDown changes the value based on a custom step', async () => {
+      const user = userEvent.setup();
+      const handleValueChange = spy();
+
+      const { getByTestId } = render(
+        <NumberInput
+          defaultValue={10}
+          step={5}
+          onValueChange={handleValueChange}
+          slotProps={{ input: { 'data-testid': 'input' } } as any}
+        />,
+      );
+
+      const input = getByTestId('input') as HTMLInputElement;
+
+      await user.click(input);
+
+      await user.keyboard('[ArrowUp]');
+      await user.keyboard('[ArrowUp]');
+      expect(handleValueChange.args[1][1]).to.equal(20);
+      expect(input.value).to.equal('20');
+
+      await user.keyboard('[ArrowDown]');
+      expect(handleValueChange.args[2][1]).to.equal(15);
+      expect(handleValueChange.callCount).to.equal(3);
+      expect(input.value).to.equal('15');
+    });
+
+    it('ArrowUp and ArrowDown changes the value based on shiftMultiplier if the Shift key is held', async () => {
+      const user = userEvent.setup();
+      const handleValueChange = spy();
+
+      const { getByTestId } = render(
+        <NumberInput
+          defaultValue={20}
+          shiftMultiplier={5}
+          onValueChange={handleValueChange}
+          slotProps={{ input: { 'data-testid': 'input' } } as any}
+        />,
+      );
+
+      const input = getByTestId('input') as HTMLInputElement;
+
+      await user.click(input);
+
+      await user.keyboard('{Shift>}[ArrowUp]/');
+      expect(handleValueChange.callCount).to.equal(1);
+      expect(handleValueChange.args[0][1]).to.equal(25);
+      expect(input.value).to.equal('25');
+
+      await user.keyboard('{Shift>}[ArrowDown][ArrowDown]{/Shift}');
+      expect(handleValueChange.args[2][1]).to.equal(15);
+      expect(handleValueChange.callCount).to.equal(3);
+      expect(input.value).to.equal('15');
+    });
+
+    it('PageUp and PageDown changes the value based on shiftMultiplier', async () => {
+      const user = userEvent.setup();
+      const handleValueChange = spy();
+
+      const { getByTestId } = render(
+        <NumberInput
+          defaultValue={20}
+          shiftMultiplier={5}
+          onValueChange={handleValueChange}
+          slotProps={{ input: { 'data-testid': 'input' } } as any}
+        />,
+      );
+
+      const input = getByTestId('input') as HTMLInputElement;
+
+      await user.click(input);
+
+      await user.keyboard('[PageUp]');
+      expect(handleValueChange.args[0][1]).to.equal(25);
+      expect(input.value).to.equal('25');
+
+      await user.keyboard('[PageDown][PageDown]');
+      expect(handleValueChange.args[2][1]).to.equal(15);
+      expect(handleValueChange.callCount).to.equal(3);
+      expect(input.value).to.equal('15');
+    });
+
+    it('sets value to max when Home is pressed', async () => {
```

Comment:
Does this behavior follow any recommendation or a spec? The native number input does not handle Home/End keys in any special way and lets them control the cursor.

Reply:
Yep ~ It's from the [Spinbutton pattern](https://www.w3.org/WAI/ARIA/apg/patterns/spinbutton/#keyboardinteraction) from the WAI ARIA APG site